qcodes/utils/plotting.py

```diff
@@ -42,4 +67,180 @@ def auto_range_iqr(data_array: np.ndarray) -> Tuple[float, float]:
+                             color_under: Optional[Any]=DEFAULT_COLOR_UNDER
+                             ) -> None:
+    """
+    Applies limits to colorscale and updates extend.
+
+    This function applies the limits `new_lim` to the heatmap plot associated
+    with the provided `colorbar`, updates the colorbar limits, and also adds
+    the colorbar clipping indicators in form of small triangles on the top and
+    bottom of the colorbar, according to the where the limits are exceeded.
+
+    Args:
+        colorbar: The actual colorbar to be updated
+        new_lim: 2-tuple of the desired minimum and maximum value of the color
+            scale. If any is `None` it will be left unchanged.
+        data_lim: 2-tuple of the actual minimum and maximum value of the data.
+            If left out the minimum and maximum are deduced from the provided
+            data, or the data associated with the colorbar.
+        data_array: numpy array containing the data to be considered for
+            scaling. Must be left out if `data_lim` is provided. If neither is
+            provided the data associated with the colorbar is used.
+        color_over: Matplotlib color representing the datapoints clipped by the
+            upper limit
+        color_under: Matplotlib color representing the datapoints clipped by
+            the lower limit
+    """
+    # browse the input data and make sure that `data_lim` and `new_lim` are
+    # available
+    if not isinstance(colorbar.mappable, matplotlib.collections.QuadMesh):
+        raise RuntimeError('Can only scale mesh data, but received '
+                           f'"{type(colorbar.mappable)}" instead')
+    if data_lim is None:
+        if data_array is None:
+            data_array = colorbar.mappable.get_array()
+        data_lim = np.nanmin(data_array), np.nanmax(data_array)
+    else:
+        if data_array is not None:
+            raise RuntimeError('You cannot specify `data_lim` and `data_array` '
+                               'please read the docstring of '
+                               '`apply_color_scale_limits:\n\n`'
+                               + apply_color_scale_limits.__doc__)
+        else:
+            data_lim = cast(Tuple[float, float], tuple(sorted(data_lim)))
+    # if `None` is provided in the new limits don't change this limit
+    vlim = [new or old for new, old in zip(new_lim, colorbar.get_clim())]
+    # sort limits in case they were given in a wrong order
+    vlim = sorted(vlim)
+    # detect exceeding colorscale and apply new limits
+    exceeds_min, exceeds_max = (data_lim[0] < vlim[0],
+                                data_lim[1] > vlim[1])
+    if exceeds_min and exceeds_max:
+        extend = 'both'
+    elif exceeds_min:
+        extend = 'min'
+    elif exceeds_max:
+        extend = 'max'
+    else:
+        extend = 'neither'
+    colorbar.extend = extend
+    # the colorbar has no setter method and setting the colorbar extend does
```

Comment:
shall we extract it this as a separate method, so that it does not disturb the main logic of the function? also, when matplotlib fixes it, it will be (yeah, just a tiny bit) easier for use to adapt the code ))